sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp

```diff
@@ -276,6 +297,24 @@ namespace Azure { namespace Core { namespace Test {
           Sanitize(testNameInfo->test_suite_name()), Sanitize(testNameInfo->name()));
       m_testContext.RecordingPath = recordingPath;
       m_interceptor = std::make_unique<Azure::Core::Test::InterceptorManager>(m_testContext);
+
+      if (!m_testContext.IsPlaybackMode())
+      {
+        auto SetBuiltinEnvironment
+            = [](std::string const& serviceName, std::string const& targetVariable) {
+                std::string targetValue = Azure::Core::_internal::Environment::GetVariable(
+                    (serviceName + targetVariable).c_str());
```

Comment:
I kept thinking about this `serviceName` inpur argument.

What if we get rid of it and we just introduce a new environment variable called `AZURE_ENV_PREFIX` or something like that and we use it instead of this input args.  

That would make test-implementation agnostic of it and only the user/developer and the test-fw would use it for what you are doing here.

Then, we can set that env var on the internal ci pipeline for live test. The internal pipeline knows the service names as it is given in the ci.yaml file.

This would also allow developers running tests for more than one service to keep using the same ENV VARS for all services instead of setting more variables.

Reply:
Doesn't this solution (adding new environment variables) require modifying new-testresources.ps1 in eng\common to emit them in the tests? That seems highly intrusive and unlikely to pass muster with the engineering system team.

Note that my change does *not* modify the AZURE_XXX variables if the service specific variable isn't set, which means that people working in multiple services will continue to be able to use the AZURE_XXX variables.

Because of your concerns, I'll update the SeUpTestBase function such that if AZURE_XXX is already set, the code does nothing. That allows you to locally set AZURE_XXX in your environment and have it work.

Reply:
> Doesn't this solution (adding new environment variables) require modifying new-testresources.ps1 in eng\common

No.  The new env var is added to `sdk/attestation/ci.yml` in the `TestEnv` section.  And it replaces the input arg for the SetUpTestBase().  It would check for the env var instead of checking the inputArg.

And the Servive name is already in the ci.yml as the `ServiceDirectory`

Reply:
Interesting idea. It's not nearly as discoverable as adding a parameter to SetUpTestBase(), but I can make that solution work.

Reply:
There's a problem. The TestEnv section from ci.yml is not provided to live tests. There doesn't appear to be a mechanism to specify environment variables to the ctest template :(.

So while this is an incredibly attractive option, it doesn't appear to be possible.

Reply:
Hmm. I might have a solution

Reply:
ci.yml pass the parameter `ServiceDirectory` to `eng/pipelines/templates/stages/archetype-sdk-client.yml` and this one pass the parameter later to `/eng/pipelines/templates/jobs/archetype-sdk-tests.yml` for liveTests.

All you need is to add: `AZURE_ENV_VAR_PREFIX: ${{ parameters.ServiceDirectory }}` to line 150:
![image](https://user-images.githubusercontent.com/24213737/155577687-7eebae64-29af-4519-abdf-64f47dfedbe6.png)


This will set the service name to that env var.  Then you read that env var from the test-fw

Reply:
Note:  `ServiceDirectory` is lowercase. We need to uppercase it.

Reply:
That is essentially my "I might have a solution". However from what I've seen, adding it to variables isn't sufficient - you need to add it to the `env:` section of the `ctest` task to ensure that it is set when ctest runs.

And IMHO the environment variable should reflect the name of the parameter - It's refered to as "ServiceDirectory" in every other part of the ES (and in every other language), so it becomes AZURE_SERVICE_DIRECTORY. That way people who are working on the ES will understand the construct.

Reply:
PS: When you reply to a resolved conversation, please unresolve it - it took me almost 10 minutes to find this comment (I *hate* the github pull request UX).

Reply:
> so it becomes AZURE_SERVICE_DIRECTORY

I am OK with any name :) .  I was suggesting something to self-describe what it will do when it is set.  But we can add documentation notes somewhere (maybe in the test fw) to describe this env var meaning

Reply:
Do you mean something like this existing comment:
```C++
     * @note If AZURE_TENANT_ID, AZURE_CLIENT_ID, or AZURE_CLIENT_SECRET are not available in the
     * environment, the AZURE_SERVICE_DIRECTORY environment variable is used to set those values
     * with the values emitted by the New-TestResources.ps1 script.
```

Reply:
Btw, for completeness: I confirmed that `variables` are added to the environment for the test, so I've added it to the variables section in `archetype-sdk-tests.yml`.